lib/pbench/server/database/models/users.py

```diff
@@ -146,11 +146,11 @@ def is_admin(self) -> bool:
         return self.role is Roles.ADMIN
 
     @staticmethod
-    def is_admin_username(username) -> bool:
-        """Determine if the given user name is the offical admin user.
+    def is_admin_username(username: str) -> bool:
+        """Determine if the given user name is an admin user.
 
         Returns:
-            True if the user name is "admin", False otherwise.
+            True if the user is an admin; False otherwise.
```

Comment:
Oh, fine; I would have preferred just leaving it as a description of what the method *does* rather than what it was envisioned to eventually become. The implementation isn't going to change; but on the other hand what we say doesn't really matter as it'll go away as the entire model becomes obsolete with Keycloak.

Reply:
I'll do whatever you two agree on, I just don't want to keep flipping back and forth.

Reply:
I approved. I probably should have pre-resolved this comment. I'd prefer that the docstring describe what it does rather than what it never will do; but it's "close enough" and it won't last long, so let's not worry about it.

Reply:
I prefer that the docstring describe what the function does (or is intended to do) _from a caller perspective_ (i.e., what the the behavior of the **interface** is).  It can also describe what the implementation does if it is necessary for readers/reviewers to understand it, but that sort of thing is probably better relegated to block comments when it is really needed.

I believe that our intention/expectation here was that we would be replacing the guts of this function but not necessarily replacing the interface to it, so that code layered on top of it would continue to work when we added groups and/or changed the definition of "admin".  Thus, I think Peter's current changes are appropriate.

Reply:
Note that the *name* is not anything that might imply "user has admin access": it's very explicitly `is_admin_username`, so I think this discussion about "intent" and internal refactoring is somewhat irrelevant. It does exactly what the name says, and if we changed that we'd need to change the name as well. Now the documentation doesn't match the name or the implementation.

Nevertheless, this is old news and I don't really care whether it's properly documented in its final moments of life, so let's move on. 😁